akka-http-cors/src/main/scala/ch/megard/akka/http/cors/scaladsl/settings/CorsSettings.scala

```diff
@@ -119,6 +119,28 @@ abstract class CorsSettings extends javadsl.settings.CorsSettings {
 
 object CorsSettings {
 
+  val prefix = "akka.http.cors"
+
+  def apply(config: Config) = fromSubConfig(config.getConfig(prefix))
+
+  def fromSubConfig(config: Config) = CorsSettings.Default(
+    allowGenericHttpRequests = config.getBoolean("allow-generic-http-requests"),
+    allowCredentials = config.getBoolean("allow-credentials"),
+    allowedOrigins = config.getStringList("allowed-origins").asScala match {
+      case l if l.size == 1 && l(0) == "*" => HttpOriginRange.*
+      case l => HttpOriginRange(l.map(HttpOrigin(_)): _*)
+    },
+    allowedHeaders = config.getStringList("allowed-headers").asScala match {
+      case l if l.size == 1 && l(0) == "*" => HttpHeaderRange.*
+      case l => HttpHeaderRange(l: _*)
+    },
+    allowedMethods = config.getStringList("allowed-methods").asScala.map(HttpMethod.custom(_)).toList,
```

Comment:
HttpMethod must be parsed into their specific instances, and fallback to `custom` only when not found.
Look at `HttpMethods` and more speficicaly at `HttpMethods.getForKey()`.